docs/contributor/arch/001-otel-collector-metric-based-pipeline-status.md

```diff
@@ -0,0 +1,102 @@
+Inevitably, interruptions can occur, potentially hindering the successful delivery of data. 
+To enhance short-term resilience, we implement retries and buffering mechanisms. 
+Nevertheless, there may be scenarios where data cannot be delivered, requiring user notification for prompt action.
+
+The typical situations causing problems are:
+
+* Connectivity problems with the third-party service.
+* Backpressure or throttling due to overloading the third-party service.
+* Reaching the ingestion limit of a pipeline, which may occur even with auto-scaling in place (subject to a maximum replica setting).
+
+These situations can be monitored by collecting metrics from the relevant components, with [recently added documentation](https://github.com/kyma-project/telemetry-manager/pull/423) to guide the process. 
+However, this approach can be cumbersome for users, as it requires them to identify which services to scrape and filter the relevant metrics. 
+Additionally, these details delve into internal aspects that may undergo changes in the future.
+
+### Goal
+
+* The telemetry manager handles the lifecycle of pipelines and should be the only entity responsible for interpreting the metrics generated by the components.
+In the event of any problematic situation, the telemetry manager should promptly report it by marking the pipeline and module status as a warning. This way, users can easily detect and address any issues.
+* Additionally, we aim to make custom metrics accessible through a dedicated endpoint, providing users with only the pertinent metrics. This endpoint will be maintained over the long term, even as internal configurations may change.
+* Another effective means of notifying users about problems is by emitting Kubernetes (k8s) events when a pipeline transitions into an unhealthy mode.
+
+### Metric Analysis
+
+In the context of metric analysis, let's consider the following scenarios:
+
+#### Direct Endpoint Scraping by the Operator (Without TSDB)
+
+While direct endpoint scraping is the simplest approach and doesn't require Prometheus integration, it comes with a lot of limitations.
+When scraping an endpoint, you receive one value per timestamp, which may suffice for gauges. However, even for commonly used metrics like rates, you would require multiple values aggregated over time.
+Additionally, Prometheus automatically handles adjustments for breaks in monotonicity, such as counter resets resulting from target restarts. Manual scraping lacks this crucial feature.
+
+#### Prometheus as a Sidecar
+
+Now that we understand the necessity of using Prometheus for even basic metric analysis, let's explore how we can deploy it.
+One deployment approach involves deploying Prometheus as a sidecar alongside the operator.
+This method remains lightweight, as it doesn't necessitate any changes in the operator's code and can be easily achieved only using kustomize.
```

Comment:
It also might ease security considerations since Prometheus and the operator share a network namespace, so Prometheus can listen on localhost and no further NetworkPolicies, etc. are required.